CS483/CS483/Kartaclysm/Components/ComponentRacer.cpp

```diff
@@ -0,0 +1,64 @@
+//----------------------------------------------------------------------------
+// ComponentRacer
+// Author: Matthew White
+//
+// Component that handles racer logic
+//----------------------------------------------------------------------------
+
+#include "ComponentRacer.h"
+
+namespace Kartaclysm
+{
+	ComponentRacer::ComponentRacer(
+		HeatStroke::GameObject* p_pGameObject)
+		:
+		Component(p_pGameObject),
+		m_pDriver(nullptr),
+		m_pKart(nullptr),
+		m_iCurrentLap(0),
+		m_iCurrentPosition(1),
+		m_iCurrentTrackPiece(0)
+	{
+		std::function<void(const HeatStroke::Event*)>* lapCompleteDelegate = new std::function<void(const HeatStroke::Event*)>(std::bind(&ComponentRacer::FinishLap, this, std::placeholders::_1));
+		HeatStroke::EventManager::Instance()->AddListener("RacerCompletedLap", lapCompleteDelegate);
+	}
+
+	ComponentRacer::~ComponentRacer()
+	{
+	}
+
+	HeatStroke::Component* ComponentRacer::CreateComponent(
+		HeatStroke::GameObject* p_pGameObject,
+		tinyxml2::XMLNode* p_pBaseNode,
+		tinyxml2::XMLNode* p_pOverrideNode)
+	{
+		return new ComponentRacer(p_pGameObject);
+	}
+
+	void ComponentRacer::Update(const float p_fDelta)
+	{
+	}
+
+	void ComponentRacer::SetKart(HeatStroke::GameObject* p_pKart)
+	{
+		m_pKart = p_pKart;
+		m_pGameObject->AddChild(m_pKart);
+	}
+
+	void ComponentRacer::SetDriver(HeatStroke::GameObject* p_pDriver)
+	{
+		m_pDriver = p_pDriver;
+		m_pGameObject->AddChild(m_pDriver);
+	}
+
+	void ComponentRacer::FinishLap(const HeatStroke::Event* p_pEvent)
+	{
+		std::string strRacerId = "";
+		p_pEvent->GetRequiredStringParameter("racerId", strRacerId);
```

Comment:
Minor thing, but we do have a "GetRequiredGameObjectParameter" method to help distinguish from regular strings and GameObject GUID's.